src/coreclr/System.Private.CoreLib/src/System/Delegate.CoreCLR.cs

```diff
@@ -130,12 +139,7 @@ public override bool Equals([NotNullWhen(true)] object? obj)
                 // fall through method handle check
             }
 
-            // method ptrs don't match, go down long path
-
-            if (_methodBase is MethodInfo && d._methodBase is MethodInfo)
-                return _methodBase.Equals(d._methodBase);
-            else
-                return InternalEqualMethodHandles(this, d);
+            return MethodDesc == d.MethodDesc;
         }
 
         public override int GetHashCode()
```

Comment:
Could `GetHashCode` use `MethodDesc` instead of `GetType()` here for better hashcodes now?

Reply:
I think so.

Reply:
Done.